src/Helper/HttpProxy.php

```
@@ -0,0 +1,84 @@
+
+        /** @var ResponseInterface $backendResponse */
+        $backendResponse = $this->client->request('GET', $uri, [
+            'headers' => array_filter([
+                'Accept' => $request->headers->get('Accept'),
+                'Cache-Control' => $request->headers->get('Cache-Control'),
+                'If-Modified-Since' => $request->headers->get('If-Modified-Since'),
+                'If-None-Match' => $request->headers->get('If-None-Match'),
+                'Referer' => $request->headers->get('Referer'),
+                'X-Forwarded-For' => implode(', ', $xForwardedFor),
+                'X-Forwarded-Host' => $request->getHost(),
+                'X-Forwarded-Port' => $request->getPort(),
+                'X-Forwarded-Proto' => $request->getScheme(),
+            ]),
+            ['http_errors' => false],
+        ]);
+
+        switch ($backendResponse->getStatusCode()) {
+            case Response::HTTP_OK:
+                $stream = $backendResponse->getBody();
+
+                $response = new StreamedResponse(
+                    function () use ($stream) {
+                        if (ob_get_length()) {
+                            ob_end_clean();
+                        }
+                        while (!$stream->eof()) {
+                            echo $stream->read(1024);
+                            flush();
+                        }
+                        $stream->close();
+                    },
+                    $backendResponse->getStatusCode()
+                );
+                break;
+            case Response::HTTP_NOT_MODIFIED:
+                $response = new Response('', $backendResponse->getStatusCode());
+                break;
+            case Response::HTTP_NOT_FOUND:
+            case Response::HTTP_GONE:
+                return new Response('', $backendResponse->getStatusCode());
+            default:
+                return new Response('', Response::HTTP_BAD_GATEWAY);
```

Better to throw an exception so can be templated.